server/cluster/cluster_worker.go

```diff
@@ -45,6 +45,9 @@ func (c *RaftCluster) HandleRegionHeartbeat(region *core.RegionInfo) error {
 
 // HandleAskSplit handles the split request.
 func (c *RaftCluster) HandleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSplitResponse, error) {
+	if c.GetUnsafeRecoveryController() != nil && c.GetUnsafeRecoveryController().IsRunning() {
```

Comment:
Do we need a lock?

Reply:
Added a lock inside the `IsRunning` method. As for the call site, I don't think we need a lock. `unsafeRecoveryController` is created once and never get destroyed, If you were worrying about the race condition here.